subprojects/core/src/main/java/org/gradle/api/internal/AbstractNamedDomainObjectContainer.java

```diff
@@ -64,6 +65,7 @@ public T create(String name, Closure configureClosure) {
 
     public T create(String name, Action<? super T> configureAction) throws InvalidUserDataException {
         assertCanAdd(name);
+        NameValidator.validate(name);
```

Comment:
Introduced `AbstractValidatingNamedDomainObjectContainer` and used that only for:
- Configurations
- Flavors
- Artifact Types